functions.inc/class.backup.php

```diff
@@ -276,14 +276,7 @@ function add_items() {
 									. '.' . backup__($x);
 						}
 					}
-					$cmd[] = ' --opt --skip-comments --skip-extended-insert --lock-tables=false --skip-add-locks --compatible=no_table_options --default-character-set=utf8';
```

Comment:
> Write INSERT statements using multiple-row syntax that includes several VALUES lists. This results in a smaller dump file and speeds up inserts when the file is reloaded. so this makes the backups faster but slows down the restore.  Is that an acceptable tradeoff

Reply:
I don't think `--skip-extended-insert` actually makes backups faster. My assumption is that it was put there because the old PEAR code couldn't handle the massive queries generated by the default settings (which enable `--extended-insert`.) I wasn't specifically checking for backup times during my testing, but I didn't notice any difference between the existing options and my proposed ones.

From a user perspective, backups will be happening unattended and automatically, while restores will be done by someone who's staring at the web UI, possibly in a panic because their PBX is offline! The magnitude of the speedup I saw (2.25 hours down to 15 seconds) is such that, IMHO, there would have to be a **very** big increase in backup time before it became an unacceptable tradeoff.

Reply:
I don't know the history behind the decision command arguments. I actually had no idea what that flag did which is why I looked at the manual for it.

The idea of mysql dumps as a backup method is kind of awful.  To do them properly you have to lock tables. If you lock a necessary table on a live system you could implode the universe.  Going faster (even by milliseconds) could be the difference between success and failure.  I think the ultimate issue is writes happening during the dump.  Short of locking speed is your only friend with the current methods which is why they will be used with extreme care in the future methods. 

I am generally ok with these changes noting my concern that it may still be an issue on systems that are larger or have a heavy call volume.

Reply:
On that note, the original options included `--lock-tables=false` which I removed but I've reverted the change, as you say it could cause problems if a backup is running during a heavy usage period. If tables were migrated to InnoDB, the `--single-transaction` option would be a perfect solution to the table locking problem.

Reply:
"If tables were migrated to InnoDB" that would be a mess. However we are close to using doctrine almost completely for everything which means new installs will be innodb

Reply:
Following up on speed concerns, here are times before and after my patch. CDR table holds 164552 records, biggest table in A2Billing database holds 174947 records. Slight increase in times to dump FreePBX database but a decrease with CDR database. I'd want to test on more systems before I draw any conclusions but it seems not to be a problem.

Database | Time to dump
----------|--------------
asterisk | 1.03439998627
asteriskcdrdb | 3.58890008926
mya2billing | 2.79249978065

Database | Time to dump
----------|--------------
asterisk | 1.66269993782
asteriskcdrdb | 3.03190016747
mya2billing | 3.7243001461